projects/interfaces/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+type OurByteBuffer struct {
+	myBuff []byte
+}
+
+// Is there a maximum about of data an OurByteBuffer can store?
+// There is no maximum as far as our memory doesn't run out.
+// As far as I learned, buffer (bytes.buffer) can grow magically and always is the same buffer.
+// But OurByteBuffer created as a slice bytes and can grow but always not the same slice bytes.
+
+// Are there any important performance characteristics? e.g. is it much faster or slower to Write one large amount of data to it than write the same amount of data one byte at a time?
+// yes, for buffer (bytes.buffer) is always faster because working with block of data.
+// for OurByteBuffer which is a slice, it is better to append a large amount of data instead of one byte at a time to trigger the copy process less or create a slice with estimated size we need.
```

Comment:
This looks good! Exactly - we want to give hints to `append` to either know how many bytes we're copying up front, or to create a slice with the right size :)